src/realm-settings/PartialImport.tsx

```diff
@@ -278,107 +285,258 @@ export const PartialImportDialog = (props: PartialImportProps) => {
+                      "clientRoles",
+                      t("common:clientRoles")
+                    )}
+                </DataList>
+              </StackItem>
+              <StackItem>
+                <Text>{t("selectIfResourceExists")}:</Text>
                 <Select
-                  toggleId="realm-selector"
-                  isOpen={isRealmSelectOpen}
-                  onToggle={() => setIsRealmSelectOpen(!isRealmSelectOpen)}
-                  onSelect={handleRealmSelect}
-                  placeholderText={targetRealm.realm || targetRealm.id}
+                  isOpen={isCollisionSelectOpen}
+                  direction="up"
+                  onToggle={() => {
+                    setIsCollisionSelectOpen(!isCollisionSelectOpen);
+                  }}
+                  onSelect={handleCollisionSelect}
+                  placeholderText={t(collisionOption)}
                 >
-                  {realmSelectOptions()}
+                  {collisionOptions()}
                 </Select>
               </StackItem>
-            )}
-            <StackItem>
-              <Text>{t("chooseResources")}:</Text>
-              <DataList aria-label="Resources to import" isCompact>
-                {targetHasResource("users") &&
-                  resourceDataListItem("users", "users")}
-                {targetHasResource("groups") &&
-                  resourceDataListItem("groups", "groups")}
-                {targetHasResource("clients") &&
-                  resourceDataListItem("clients", "clients")}
-                {targetHasResource("identityProviders") &&
-                  resourceDataListItem(
-                    "identityProviders",
-                    "identity providers"
-                  )}
-                {targetHasRealmRoles() &&
-                  resourceDataListItem("realmRoles", "realm roles")}
-                {targetHasClientRoles() &&
-                  resourceDataListItem("clientRoles", "client roles")}
-              </DataList>
-            </StackItem>
-            <StackItem>
-              <Text>{t("selectIfResourceExists")}:</Text>
-              <Select
-                isOpen={isCollisionSelectOpen}
-                direction="up"
-                onToggle={() => {
-                  setIsCollisionSelectOpen(!isCollisionSelectOpen);
-                }}
-                onSelect={handleCollisionSelect}
-                placeholderText={t(collisionOption)}
-              >
-                {collisionOptions()}
-              </Select>
-            </StackItem>
-          </>
-        )}
-      </Stack>
-    </Modal>
+            </>
+          )}
+        </Stack>
+      </Modal>
+    );
+  };
+
+  const importCompleteMessage = () => {
+    return t("importSuccess", {
+      added: importResponse?.added,
+      overwritten: importResponse?.overwritten,
+      skipped: importResponse?.skipped,
+    });
+  };
+
+  const loader = async (first?: number, max?: number) => {
+    let last = first! + max!;
```

Comment:
Using the [non-null assertion operator](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator) (`!`) here can cause exceptions at run-time as the value can be possibly `undefined`. When using this operator you are essentially saying to the compiler that it should not be catching these errors at compile time as you are 'sure' that this is fine.

A better solution here I think would be to provide some default values so that we're not disabling any compile-time checks.
```suggestion
  const loader = async (first = 0, max = 15) => {
    let last = first + max;
```